controller/user_blackbox_test.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/jinzhu/gorm"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UserTestSuite struct {
+	gormtestsupport.DBTestSuite
+}
+
+func (s *UserTestSuite) SetupTest() {
+	s.DBTestSuite.SetupTest()
+}
+
+func TestRunUserREST(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &UserTestSuite{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *UserTestSuite) SecuredController(user *auth.Identity) (*goa.Service, *UserController) {
+	svc, err := auth.ServiceAsUser("User-Service", user)
+	require.NoError(s.T(), err)
+	return svc, NewUserController(svc, s.Application)
+}
+func (s *UserTestSuite) UnsecuredController() (*goa.Service, *UserController) {
+	svc := goa.New("User-Service")
+	controller := NewUserController(svc, s.Application)
+	return svc, controller
+}
+
+func (s *UserTestSuite) TestShowClusterAvailableToUser() {
+
+	s.T().Run("ok", func(t *testing.T) {
+
+		t.Run("empty", func(t *testing.T) {
+			// given
+			identity := auth.NewIdentity()
+
+			// when
+			svc, userCtrl := s.SecuredController(identity)
+			_, clusters := test.ClustersUserOK(t, svc.Context, svc, userCtrl)
+
+			// then
+			assert.NotNil(t, clusters)
+			assert.Empty(t, clusters.Data)
+
+		})
+
+		t.Run("list single cluster", func(t *testing.T) {
+			// given
+			identity := auth.NewIdentity()
+
+			cl := testsupport.CreateCluster(t, s.DB)
+			identityCluster := testsupport.CreateIdentityCluster(t, s.DB, cl, &identity.ID)
+			require.NotNil(t, identityCluster)
+
+			// when
+			svc, userCtrl := s.SecuredController(identity)
+			_, clusters := test.ClustersUserOK(t, svc.Context, svc, userCtrl)
+
+			// then
+			require.NotNil(t, clusters)
+			require.NotNil(t, clusters.Data)
+
+			testsupport.AssertEqualClusterData(t, []repository.Cluster{*cl}, clusters.Data)
+		})
+
+		t.Run("list multiple cluster", func(t *testing.T) {
+			// given
+			expectedClusters := make([]repository.Cluster, 0)
+			// create cluster identity for random cluster
+			identity := auth.NewIdentity()
+			identityID := identity.ID
+
+			c := testsupport.CreateCluster(t, s.DB)
+			expectedClusters = append(expectedClusters, *c)
+
+			identityCluster := testsupport.CreateIdentityCluster(t, s.DB, c, &identityID)
+			require.NotNil(t, identityCluster)
+
+			// create cluster identity for cluster loading from config
+			err := s.Application.ClusterService().CreateOrSaveClusterFromConfig(context.Background())
+			require.NoError(s.T(), err)
+
+			osoClusters, err := s.Application.Clusters().Query(func(db *gorm.DB) *gorm.DB {
+				return db.Where("type = ?", cluster.OSO)
+			})
+
+			for _, c := range osoClusters {
+				cl := c
+				identityCluster := testsupport.CreateIdentityCluster(t, s.DB, &cl, &identityID)
+				require.NotNil(t, identityCluster)
+				expectedClusters = append(expectedClusters, cl)
+			}
```

Comment:
Why just not repeat lines 94 - 98 a few times in a loop instead of lines 100 - 113? We are not testing configuration loading here, are we? Then the test won't rely on the configuration implementation which is not relevant here.

Reply:
Here we are not testing configuration loading, but I wanted to test some clusters already in DB which is loaded from configuration while it has a relationship to identity and some cluster are created by doing post call to `/cluster` api which is not present in the cluster config. So in this way we have both type of cluster while showing it to user

Reply:
But why this test cares about how the clusters were created? You test the control er which loads stuff from the DB. Not creating.
I would avoid creating unnecessary dependencies here. For easier test maintenance sake. If we change configuration loading then we won't need to update his test. Only the corresponding configuration tests.